oidc-controller/api/main.py

```diff
@@ -97,41 +97,46 @@ async def logging_middleware(request: Request, call_next) -> Response:
     try:
         response: Response = await call_next(request)
         return response
-    finally:
+    except RedisCriticalError:
+        # Don't catch Redis critical errors - let them crash the app
+        raise
+    except Exception as e:
         process_time = time.time() - start_time
-        # If we have a response object, log the details
-        if "response" in locals():
-            logger.info(
-                "processed a request",
-                status_code=response.status_code,
-                process_time=process_time,
-            )
-        # Otherwise, extract the exception from traceback, log and return a 500 response
-        else:
-            logger.info(
-                "failed to process a request",
-                status_code=http_status.HTTP_500_INTERNAL_SERVER_ERROR,
-                process_time=process_time,
-            )
-
-            # Need to explicitly log the traceback
-            logger.error(traceback.format_exc())
-
-            return JSONResponse(
-                status_code=http_status.HTTP_500_INTERNAL_SERVER_ERROR,
-                content={
-                    "status": "error",
-                    "message": "Internal Server Error",
-                    "process_time": process_time,
-                },
-            )
+        logger.info(
+            "failed to process a request",
+            status_code=http_status.HTTP_500_INTERNAL_SERVER_ERROR,
+            process_time=process_time,
+        )
+
+        # Need to explicitly log the traceback
+        logger.error(traceback.format_exc())
+
+        return JSONResponse(
+            status_code=http_status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content={
+                "status": "error",
+                "message": "Internal Server Error",
+                "process_time": process_time,
+            },
+        )
+    else:
+        process_time = time.time() - start_time
+        logger.info(
+            "processed a request",
+            status_code=response.status_code,
+            process_time=process_time,
+        )
 
 
 @app.on_event("startup")
 async def on_tenant_startup():
     """Register any events we need to respond to."""
     await init_db()
     await init_provider(await get_db())
+
+    # Validate Redis connection if USE_REDIS_ADAPTER is enabled
+    await validate_redis_connection()
```

Comment:
This part's good to check I think, no point in vcauth process existing if it needs Redis and it's not there, as long as this isn't some race condition stuff on deployment where the Redis is still starting and the vcauth pod then keeps trying to start up and then crashing out constantly because it's not ready